### PR TITLE
Bump to SDK 30

### LINKED ISSUE #1363: target API level 30 (due November 2021)
Android Talk has to target API level 30 before November 2021:

grafik

https://developer.android.com/distribute/best-practices/develop/target-sdk

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

resolves #1363

### CODE REVIEW

**Comment by timkrueger** [comment]
This PR relates to issue #1363

**Comment by timkrueger** [comment]
To document the progress for errors and warnings @AndyScherzinger  mentioned:

- [x]  `WRITE_EXTERNAL_STORAGE`
- [ ]  `AudioManager` constants

File: app/src/main/java/com/nextcloud/talk/jobs/UploadAndShareFilesWorker.kt [hunk 1]
```diff
@@ -213,31 +213,64 @@ class UploadAndShareFilesWorker(val context: Context, workerParameters: WorkerPa
         const val META_DATA = "META_DATA"
 
         fun isStoragePermissionGranted(context: Context): Boolean {
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```
Comment (AndyScherzinger) [inline app/src/main/java/com/nextcloud/talk/jobs/UploadAndShareFilesWorker.kt:216]: wouldn't this still have to be checked for >=M && <30 ?

Comment (AndyScherzinger) [reply app/src/main/java/com/nextcloud/talk/jobs/UploadAndShareFilesWorker.kt:216]: nevermind, I just now saw the check below :)

**Comment by timkrueger** [comment]
Are my solutions in commits e1d9afa1a26ef09970b90021eb13be63c3eb4e5f and e1d9afa1a26ef09970b90021eb13be63c3eb4e5f OK?

//cc @AndyScherzinger , @mahibi

**Comment by AndyScherzinger** [comment]
> Are my solutions in commits e1d9afa and e1d9afa OK?

I'd say so, yes. Not sure why this is marked as error, probably because in theory you could have manipulated the int value somewhere so it is not matching the values of the AudioManager. So seems all good in our case 👍

**Comment by AndyScherzinger** [comment]
Also it seems the ignore values aren't the right ones for the constants. ;)

**Comment by AndyScherzinger** [comment]
suppress lines seem to work, it seems it just took ages for the results to be posted... sry for the noise

**Review by mahibi — APPROVED** [review]

**Review by AndyScherzinger — APPROVED** [review]

**Pull request merged**